### PR TITLE
feat(performance): Add team key transaction button to transaction summary

### PR DESCRIPTION
**Pull request by Zylphrex**
This replaces the key transaction button on the transaction summary page with a
team based key transaction when the `team-key-transactions` flag is enabled.

### CODE REVIEW

File: static/app/views/performance/transactionSummary/teamKeyTransactionButton.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+type Props = {
+  eventView: EventView;
+  organization: Organization;
+  teams: Team[];
+  transactionName: string;
+};
+
+function TeamKeyTransactionButton({eventView, teams, ...props}: Props) {
+  if (eventView.project.length !== 1) {
```
Comment (k-fish) [inline static/app/views/performance/transactionSummary/teamKeyTransactionButton.tsx:21]: `project = [-1]`? I also wish that property wasn't singular lol.

As an aside we should probably make a 'isSingleProject' util function or function on the EventView.

**Review by k-fish — APPROVED** [review]
Mostly looks good and it's behind a flag, just had a couple small comments. Also maybe include a screenshot since this has some UI changes?

File: static/app/components/performance/teamKeyTransaction.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      });
+    }
+  }
+
+  handleToggleKeyTransaction = async (selection: TeamSelection) => {
+    // TODO: handle the max 100 limit
+    const {api, organization, project, teams, transactionName} = this.props;
+    const markAsKeyTransaction = selection.action === 'key';
+
+    let teamIds;
```
[the remark this answers is not in the document]
Comment (Zylphrex) [reply static/app/components/performance/teamKeyTransaction.tsx:100]: Ahh sorry, the optional `teamIds` in the `toggleKeyTransaction` function is for backwards compatibility so the existing key transactions functionality doesn't break. The idea is to make it required once that's removed.

File: static/app/views/performance/transactionSummary/teamKeyTransactionButton.tsx (hunk 1 above)
Comment (Zylphrex) [reply static/app/views/performance/transactionSummary/teamKeyTransactionButton.tsx:21]: Going to leave this for the time being, with the intent to remove the `eventView` prop in an upcoming PR.

**Pull request merged**